src/StreamJsonRpc/JsonRpcEventSource.cs

```diff
@@ -278,18 +278,33 @@ internal static string GetArgumentsString(object? arguments)
 
             const int maxLength = 128;
 
-            if (arguments is object?[] args)
+            if (arguments is null)
+            {
+                return string.Empty;
+            }
+            else if (arguments is object[] args)
+            {
+                var stringBuilder = new StringBuilder();
+                for (int i = 0; i < args.Length; ++i)
+                {
+                    stringBuilder.Append($"arg{i}: {args[i]}, ");
+                    if (stringBuilder.Length > maxLength)
+                    {
+                        return $"{stringBuilder.ToString(0, maxLength)}...(truncated)";
+                    }
+                }
+
+                return stringBuilder.ToString();
+            }
+            else if (arguments is Dictionary<string, object> dict)
```

Comment:
By using the type `IReadOnlyDictionary<string, object?>` I think you'll match more, and recognize that the values may be null.

Reply:
Done

Reply:
I did belatedly realize I could use `IDictionary.Keys` to do the iteration.  It's not as tidy, but it would catch even more cases.  Thoughts?

Reply:
Actually, it looks like `Dictionary<K, V>` doesn't implement `IReadOnlyDictionary<K, V>`?

Reply:
@amcasey: [it most decidedly does implement `IReadOnlyDictionary<K, V>`](https://apisof.net/catalog/589059d3-d7f8-e09b-705d-91a461971cc2). And I'm 99% sure they implemented that interface in the same release of .NET that introduced the interface.

Reply:
Odd, I'm sure that comment was based on a failure I saw at runtime, but I can't reproduce it now.  I must have done something else wrong and misinterpreted the resulting exception.